odk2-src/build-app.rst

```diff
@@ -0,0 +1,778 @@
+
+Once you have made these changes and used XLSX Converter on the :file:`framework.xlsx` file to update the :file:`app/config/assets/framework/forms/framework/formDef.json` file, you should see your new form show up in the :guilabel:`Preview` tab of the Application Designer. Clicking on that should open your form.
+
+.. tip::
+
+  If you don't see your form in the :guilabel:`Preview`, try refreshing your browser.
+
+.. _build-app-debugging-survey:
+
+Debugging your Survey
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The XLSX Converter should report most problems with your survey.
+
+If the form is not being rendered correctly but your survey generates a :file:`formDef.json` without an error, first try purging the database (dropping all the existing data tables) using the :guilabel:`Purge Database` button on the :guilabel:`Preview` tab. You will typically need to purge the database whenever you add or remove fields from your form or change their data type.
+
+If that does not resolve the issue, try stopping the :program:`grunt` command (on Windows, :kbd:`Control-C` should produce a prompt asking to confirm whether to stop or not. On Mac, :kbd:`Control-C` kill the process with no prompt.), and re-running it. :program:`Grunt` can sometimes get overwhelmed with changes and stop working. After restarting, test your form.
+
+If there are other problems, the contents of the JavaScript Console will be helpful to the ODK core team for debugging. Open the JavaScript Console by clicking the icon with the three bars in the top right, select :guilabel:`More Tools`, select :guilabel:`Developer Tools`, and then select the :guilabel:`Console` tab. Select all of the debugging output, then copy it, save it to a file, and post it to the |forum|_ or create a ticket on the `Github Issue Tracker <https://github.com/opendatakit/opendatakit/issues>`_.
+
+.. _build-app-move-to-device:
+
+Moving Files To The Device
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. note::
+  You must have USB debugging enabled on your device in order to perform this step. See `these instructions <https://www.phonearena.com/news/How-to-enable-USB-debugging-on-Android_id53909>`_ for help.
+
+In order to see these changes on an Android device, you must first have ODK Survey installed on your device. Then:
+
+  #. Connect the device to your computer via a USB cable
+  #. Open a :program:`cmd` or :program:`terminal` window within the :guilabel:`Application Designer` directory (the one containing :file:`Gruntfile.js`), as described in the :doc:`app-designer-directories` documentation.
+  #. Type:
+
+.. code-block:: console
+
+  $ grunt adbpush
```

Comment:
This code-block should be indented to put in it in the list

Reply:
It would be worth mentioning here that you need to reset configuration on the device for some changes to show up.